django-stubs/db/models/sql/query.pyi

```diff
@@ -81,51 +81,71 @@ class Query(BaseExpression):
+    ) -> tuple[list[Any], Any, tuple[Any, ...], list[str]]: ...
     def get_meta(self) -> Options: ...
     def clone(self) -> Query: ...
     def chain(self, klass: type[Query] | None = ...) -> Query: ...
     def get_count(self, using: str) -> int: ...
+    def get_group_by_cols(self, wrapper: Any | None = None) -> list[Any]: ...
     def has_filters(self) -> WhereNode: ...
+    def get_external_cols(self) -> list[Any]: ...
+    def exists(self, limit: bool = True) -> Any: ...
     def has_results(self, using: str) -> bool: ...
     def explain(self, using: str, format: str | None = ..., **options: Any) -> str: ...
     def combine(self, rhs: Query, connector: str) -> None: ...
-    def deferred_to_data(self, target: dict[Any, Any], callback: Callable) -> None: ...
     def ref_alias(self, alias: str) -> None: ...
     def unref_alias(self, alias: str, amount: int = ...) -> None: ...
     def promote_joins(self, aliases: Iterable[str]) -> None: ...
     def demote_joins(self, aliases: Iterable[str]) -> None: ...
     def reset_refcounts(self, to_counts: dict[str, int]) -> None: ...
+    def check_alias(self, alias: str) -> None: ...
+    def check_related_objects(self, field: Any, value: Any, opts: Any) -> None: ...
+    def check_query_object_type(self, value: Any, opts: Any, field: Any) -> None: ...
     def change_aliases(self, change_map: dict[str | None, str]) -> None: ...
-    def bump_prefix(self, outer_query: Query) -> None: ...
+    def bump_prefix(self, other_query: Any, exclude: Any | None = None) -> None: ...
     def get_initial_alias(self) -> str: ...
     def count_active_tables(self) -> int: ...
     def resolve_expression(self, query: Query, *args: Any, **kwargs: Any) -> Query: ...  # type: ignore[override]
     def resolve_lookup_value(
         self, value: Any, can_reuse: set[str] | None, allow_joins: bool, summarize: bool = False
     ) -> Any: ...
-    def solve_lookup_type(self, lookup: str) -> tuple[Sequence[str], Sequence[str], Expression | Literal[False]]: ...
+    def solve_lookup_type(self, lookup: str, summarize: bool = False) -> tuple[list[str], tuple[Any, ...], Any]: ...
+    def table_alias(
+        self, table_name: str, create: bool = False, filtered_relation: Any | None = None
+    ) -> tuple[str, bool]: ...
+    def get_aggregation(self, using: Any, aggregate_exprs: dict[str, Any]) -> dict[str, Any]: ...
     def build_filter(
         self,
-        filter_expr: Q | Expression | dict[str, str] | tuple[str, Any],
-        branch_negated: bool = ...,
-        current_negated: bool = ...,
-        can_reuse: set[str] | None = ...,
-        allow_joins: bool = ...,
-        split_subq: bool = ...,
-        check_filterable: bool = ...,
-        update_join_types: bool = ...,
-    ) -> tuple[WhereNode, Iterable[str]]: ...
-    def add_filter(self, filter_clause: tuple[str, Any]) -> None: ...
+        filter_expr: Any,
```

Comment:
Why did you change `filter_expr`'s type?

Reply:
The change was a mistake. I've reverted the type of filter_expr to its original state.

Reply:
The change was a mistake. I've reverted the type to its original state.